### PR TITLE
Fix highlighting of SignColumn and cursor line number

### PR DESCRIPTION
**Pull request by rkiyanchuk**
This pull requests addresses issues with highlighting the columns left of line numbering as well as cursor line number itself.

The `SignColumn` problems has been discovered in issue altercation/solarized#252.

Commits fixing that are authored by @jwhitley and found in his [forked repository](https://github.com/jwhitley/vim-colors-solarized/commits/master).

`CursorLineNr` highlighting problem has been discovered in issue #82.

For easy comparison, this is how highlighting would look like with current upstream solarized:
upstream solarized

This is how highlighting looks after the fixes:
fixed solarized

I'm not pro in vim scheming, so please let me know if further adjustments are needed, I will be glad to improve the pull request.

### CODE REVIEW

**Comment by ELLIOTTCABLE** [comment]
Is there a reason this was never merged? Is this still pending, or …

**Comment by rkiyanchuk** [comment]
@ELLIOTTCABLE, it appears @altercation has been having time scarcity lately. He however told that the project is not going to get abandoned and he'll take care of the pull requests closer to autumn. Until then I think we can just make a fork and merge some of the fixes and improvements, so @altercation can merge those back eventually, making life easier for him.

File: colors/solarized.vim [hunk 1]
```diff
@@ -654,7 +654,7 @@ exe "hi! DiffDelete"     .s:fmt_none   .s:fg_red    .s:bg_base02
 exe "hi! DiffText"       .s:fmt_none   .s:fg_blue   .s:bg_base02 .s:sp_blue
     endif
 endif
-exe "hi! SignColumn"     .s:fmt_none   .s:fg_base0
+exe "hi! SignColumn"     .s:fmt_none   .s:fg_base0  .s:bg_base03
```
Comment (andyleejordan) [inline colors/solarized.vim]: I think this should be `s:bg_back` instead so it respects settings that change the background base color.

**Comment by fyodoriv** [comment]
Using @jwhitley's fork helped me with neomake background issues as well.

**Pull request closed without merging**